tests/nn/wrap/test_wrap.py

```diff
@@ -62,6 +62,43 @@ def test_auto_wrap(self):
         assert isinstance(model.module[2].module[0], nn.Linear)
         assert isinstance(model.module[2].module[1], nn.Linear)
 
+    def test_auto_wrap_all_modules(self):
+        """
+        Setting a low min_num_params so that all modules are wrapped.
+        """
+        with enable_wrap(wrapper_cls=FSDP, process_group=self.process_group, flatten_parameters=False):
+            sequential = nn.Sequential(
+                nn.Linear(5, 5), nn.Linear(5, 5), nn.Sequential(nn.Linear(5, 5), nn.Linear(5, 5))
+            )
+            my_auto_wrap_policy = functools.partial(default_auto_wrap_policy, min_num_params=5)
```

Comment:
Maybe you have considered this option, but what if we add an option to the default_auto_wrap_policy for this behavior? Adding this option to auto_wrap would affect all policy functions that users may write for auto_wrap? Does that make policy functions harder to write because of this option?

Again, thank you so much for the PR and awesome test cases. My general feeling is that if we can limit options in auto_wrap and try to let the policy function control wrapping, that'd be better. What do you think @ruanslv ?